packages/components/src/components/trace/TraceFilter.vue

```diff
@@ -0,0 +1,354 @@
+      <span class="trace-filter__prefix">
+        <SearchIcon />
+      </span>
+      <input
+        ref="input"
+        v-model="filterValue"
+        class="trace-filter__input"
+        type="text"
+        autocomplete="off"
+        placeholder="search events..."
+        @focus="onInputFocus"
+      />
+      <span v-if="filterValue" class="trace-filter__suffix" @click="clearValue">
+        <CloseThinIcon />
+      </span>
+      <ul
+        ref="suggestionsList"
+        v-if="suggestions.length"
+        class="trace-filter__suggestions"
+      >
+        <li
+          :class="getSuggestionClasses(index)"
+          v-for="(item, index) in suggestions"
+          :key="item.id"
+          @click.stop="makeSelection(item.id)"
+        >
+          {{ item.toString() }}
+        </li>
+      </ul>
+    </form>
+    <div class="trace-filter__arrows" v-if="nodesLength">
+      <div class="trace-filter__arrow" @click="onPrevArrow">
+        <ArrowSearchLeftIcon />
+      </div>
+      <div class="trace-filter__arrows-text">
+        <span v-if="Number.isInteger(currentIndex)">
+          <b>{{ currentIndex + 1 }}</b>
+          /
+        </span>
+        {{ nodesLength }} results
+      </div>
+      <div class="trace-filter__arrow" @click="onNextArrow">
+        <ArrowSearchRightIcon />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import ArrowSearchLeftIcon from '@/assets/arrow-search-left.svg';
+import ArrowSearchRightIcon from '@/assets/arrow-search-right.svg';
+import CloseThinIcon from '@/assets/close-thin.svg';
+import SearchIcon from '@/assets/search.svg';
+import { getEventTarget } from '@appland/diagrams';
+
+export default {
+  name: 'v-trace-filter',
+
+  components: {
+    ArrowSearchLeftIcon,
+    ArrowSearchRightIcon,
+    CloseThinIcon,
+    SearchIcon,
+  },
+
+  props: {
+    nodesLength: {
+      type: Number,
+      require: false,
+      default: 0,
+    },
+    currentIndex: {
+      type: Number,
+      default: null,
+    },
+    suggestions: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  data() {
+    return {
+      filterValue: '',
+      showSuggestions: false,
+      selectedSuggestion: 0,
+    };
+  },
+
+  watch: {
+    filterValue: {
+      handler(value) {
+        this.$emit('onChange', value);
+      },
+    },
+    selectedSuggestion: {
+      handler(selectedSuggestion) {
+        const selected =
+          this.$refs.suggestionsList.querySelectorAll('li')[selectedSuggestion];
+
+        if (selected.offsetTop < this.$refs.suggestionsList.scrollTop) {
+          this.$refs.suggestionsList.scrollTop = selected.offsetTop;
+        } else if (
+          selected.offsetTop >=
+          this.$refs.suggestionsList.scrollTop +
+            this.$refs.suggestionsList.offsetHeight
+        ) {
+          this.$refs.suggestionsList.scrollTop =
+            selected.offsetTop - selected.offsetHeight * 2;
+        }
```

Comment:
There's something off about this. When hitting the down arrow key, the first item that is below the visible area is not shown. This video is recorded on Chrome 97.

https://user-images.githubusercontent.com/8737782/152190699-f9b51098-0047-4721-b2ad-4c4b2de200e8.mp4

Reply:
Fixed https://github.com/applandinc/appmap-js/pull/534/commits/17162e5790996cae1eac7ad34a92674287cc186f